code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -277,13 +280,32 @@
 		attack_threshold_check(damage,M.melee_damage_type)
 		return 1
 
-/mob/living/simple_animal/bullet_act(obj/item/projectile/Proj)
-	if(!Proj)
+/mob/living/simple_animal/bullet_act(obj/item/projectile/P)
+	if(!P)
 		return
-	apply_damage(Proj.damage, Proj.damage_type)
-	Proj.on_hit(src)
+	if(prob(mob_reflect_chance))
+		visible_message("<span class='danger'>The [P.name] gets reflected by [name]!</span>")
+		if(P.starting)
+			var/new_x = P.starting.x + pick(0, 0, 0, 0, 0, -1, 1, -2, 2)
+			var/new_y = P.starting.y + pick(0, 0, 0, 0, 0, -1, 1, -2, 2)
+			var/turf/curloc = get_turf(src)
+
+			P.original = locate(new_x, new_y, P.z)
+			P.starting = curloc
+			P.current = curloc
+			P.firer = src
+			P.yo = new_y - curloc.y
+			P.xo = new_x - curloc.x
+			P.Angle = ""
```

Comment:
it is,or was before rebase uness they changed it,i tested it before rebase and worked *shrug